### PR TITLE
Working version

### PR DESCRIPTION
**Pull request by Schnurber**
This is my last version, it works so far, maybe you can merge this and continue working with it.  Flipping does not work, also the logic of drawing the map is in `flame_tiled` and not in `tiled`, maybe one should still modify that.

### CODE REVIEW

File: example/pubspec.yaml [hunk 1]
```diff
@@ -6,10 +6,10 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
-  flame: 0.24.0
+  flame: ^1.0.0-rc8
```
Comment (spydon) [inline example/pubspec.yaml]: Can you update this to `1.0.0-releasecandidate.13`?

File: lib/tiled.dart [hunk 2]
```diff
@@ -88,73 +90,104 @@ class Tiled {
...
   Future _load() async {
     map = await _loadMap();
-    image = await Flame.images.load(map.tilesets[0].image.source);
+    //String? src = map.tilesets[0].image!.source;
+    //if (src != null) {
+    //image = await Flame.images.load(src);
     batches = await _loadImages(map);
     generate();
     _loaded = true;
+    //}
```
Comment (spydon) [inline lib/tiled.dart]: Should these comments be removed?

File: lib/tiled_component.dart [hunk 3]
```diff
@@ -19,14 +19,19 @@ class TiledComponent extends Component {
 
   @override
   void render(Canvas canvas) {
-    _tiled.render(canvas);
+    //_tiled!.generate();
```
Comment (spydon) [inline lib/tiled_component.dart]: Should this comment be removed?

File: lib/tiled_component.dart [hunk 4]
```diff
@@ -19,14 +19,19 @@ class TiledComponent extends Component {
...
   @override
-  bool loaded() => _tiled.loaded();
+  Future<void> onLoad() async {
+    await future;
+  }
+
+  bool loaded() => _tiled!.loaded();
 
-  get future => _tiled.future;
+  get future => _tiled!.future;
```
Comment (spydon) [inline lib/tiled_component.dart]: I don't think this getter is needed?

File: lib/tiled_component.dart [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+import 'package:flame/components.dart';
 import 'package:tiled/tiled.dart';
 
 import 'dart:ui';
 
 import './tiled.dart';
 
 class TiledComponent extends Component {
-  Tiled _tiled;
+  Tiled? _tiled;
```
Comment (spydon) [inline lib/tiled_component.dart]:
```suggestion
  late final Tiled _tiled;
```

Comment (spydon) [reply lib/tiled_component.dart]: And also remove all `_tiled!` exclamation marks.

File: lib/tiled_component.dart (hunk 4 above)
Comment (Schnurber) [reply lib/tiled_component.dart]: Wothout the getter, ` await tiled.future;` does not work, see `tiled_test.dart`

File: example/pubspec.yaml (hunk 1 above)
Comment (Schnurber) [reply example/pubspec.yaml]: OK!

File: lib/tiled.dart (hunk 2 above)
Comment (Schnurber) [reply lib/tiled.dart]: Yes, OK!

File: lib/tiled_component.dart (hunk 3 above)
Comment (Schnurber) [reply lib/tiled_component.dart]: OK!

File: lib/tiled_component.dart (hunk 5 above)
Comment (Schnurber) [reply lib/tiled_component.dart]: OK!

File: example/lib/main.dart [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
+import 'package:flame/components.dart';
 import 'package:flame/game.dart';
-import 'package:flutter/widgets.dart' hide Animation;
-import 'package:tiled/tiled.dart' show ObjectGroup, TmxObject;
 import 'package:flame_tiled/flame_tiled.dart';
+import 'package:flutter/widgets.dart' hide Animation, Image;
+import 'package:tiled/tiled.dart' show ObjectGroup, TiledObject;
 
 void main() {
   WidgetsFlutterBinding.ensureInitialized();
```
Comment (spydon) [inline example/lib/main.dart:10]: I don't think this is needed anymore

File: lib/tiled.dart [hunk 7]
```diff
@@ -2,79 +2,83 @@ import 'dart:math' as math;
...
 import 'package:flame/flame.dart';
-import 'package:flame/sprite_batch.dart';
+import 'package:flame/sprite.dart';
 import 'package:flutter/material.dart' show Colors;
-import 'package:tiled/tiled.dart' hide Image;
+import 'package:tiled/tiled.dart';
 
 /// Tiled represents all flips and rotation using three possible flips: horizontal, vertical and diagonal.
-/// This class converts that representation to a simpler one, that uses one angle (with pi/2 steps) and two flips (H or V).
+/// This class converts that representation to a simpler one, that uses one angle (with pi/2 steps) and two shifts (X and Y).
```
Comment (spydon) [inline lib/tiled.dart:13]: What does `shifts` mean here? It sounds like it is shifting the position, but it has something to do with flips right?

**Review by spydon — APPROVED** [review]
I guess there are some things to do left? But maybe it is worth merging this before?
Good job anyhow!

**Comment by spydon** [comment]
@Schnurber can you rebase this on `main`?

**Review by luanpotter — APPROVED** [review]
I will merge and work off this!

**Pull request merged**